src/daemon/plugin.c

```diff
@@ -1295,11 +1301,99 @@ int plugin_register_write (const char *name,
 				(void *) callback, ud));
 } /* int plugin_register_write */
 
+static int plugin_flush_timeout_callback (user_data_t *ud)
+{
+	flush_callback_t *cb = ud->data;
+
+	return plugin_flush (cb->name, cb->timeout, /* identifier = */ NULL);
+} /* static int plugin_flush_callback */
+
+static void plugin_flush_timeout_callback_free (void *data)
+{
+	flush_callback_t *cb = data;
+
+	if (cb == NULL) return;
+
+	sfree(cb->name);
+	sfree(cb);
+} /* static void plugin_flush_callback_free */
+
 int plugin_register_flush (const char *name,
 		plugin_flush_cb callback, user_data_t *ud)
 {
-	return (create_register_callback (&list_flush, name,
-				(void *) callback, ud));
+	int status;
+	plugin_ctx_t ctx = plugin_get_ctx ();
+
+	status = create_register_callback (&list_flush, name,
+		(void *) callback, ud);
+	if (status != 0)
+		return status;
+
+	if (ctx.flush_interval != 0)
+	{
+		char *flush_prefix = "flush/";
+		size_t prefix_size;
+		char *flush_name;
+		size_t name_size;
+		user_data_t ud;
+		flush_callback_t *cb;
+
+		prefix_size = strlen(flush_prefix);
+		name_size = strlen(name);
+
+		flush_name = (char *) malloc (sizeof (char) *
+			(name_size + prefix_size + 1));
+		if (flush_name == NULL)
+		{
+			ERROR ("plugin_register_flush: malloc failed.");
+			plugin_unregister (list_flush, name);
+			return (-1);
+		}
+
+		sstrncpy (flush_name, flush_prefix, prefix_size + 1);
```

Comment:
These 12 or so lines constructing `flush_name` appear again in `plugin_unregister_flush()`. How about a function like this:?

```c
static char *plugin_flush_callback_name (char const *base_name);
```

I see the following benefits:

* Less code duplication.
* Makes it harder to break by partial code changes.
* `if { … }` blocks become shorter.
* `ERROR()` reporting when `malloc()` fails can be done in the function.

I'm not super happy with the above name, as "callback" is not a very good name for what I try to describe, but I can't think of a better alternative atm.